File: src/types/order.ts
export interface OrderItem {
  product_id: string;
  product_name?: string;
  sku?: string;
  product_size?: string;
  quantity: number;
  unit_price: number;
  total?: number;
}

export interface Address {
  street_address: string;
  city: string;
  province: string;
  postal_code: string;
  country?: string;
  phone?: string;
}

export interface Order {
  id: string;
  order_number: string;
  customer_id?: string;
  customer_email: string;
  customer_phone: string;
  customer_first_name: string;
  customer_last_name?: string;
  items: OrderItem[];
  shipping_address: Address;
  billing_address: Address;
  shipping_same_as_billing: boolean;
  source: 'online' | 'pos';
  store_id?: string;
  status: 'pending' | 'confirmed' | 'processing' | 'shipped' | 'delivered' | 'cancelled' | 'refunded';
  payment_status: 'pending' | 'paid' | 'failed' | 'refunded';
  subtotal: number;
  tax_amount: number;
  shipping_amount: number;
  discount_amount: number;
  total: number;
  payment_method?: string;
  payment_transaction_id?: string;
  payment_provider?: string;
  tracking_number?: string;
  estimated_delivery_date?: Date;
  notes?: string;
  admin_notes?: string;
  is_priority: boolean;
  created_at: Date;
  updated_at: Date;
  completed_at?: Date;
}

export interface CreateOrderPayload {
  customer_email: string;
  customer_phone: string;
  customer_first_name: string;
  customer_last_name?: string;
  items: OrderItem[];
  shipping_address: Address;
  billing_address?: Address;
  tax_amount?: number;
  shipping_amount?: number;
  discount_amount?: number;
  payment_method?: string;
  notes?: string;
  customer_id?: string;
}

File: apps/pos/Models/Order.cs
using System;
using System.Collections.Generic;

namespace pos.Models;

public class Order
{
    public Guid Id { get; set; } = Guid.NewGuid();
    public string OrderNumber { get; set; } = string.Empty;
    public string CustomerEmail { get; set; } = string.Empty;
    public string CustomerPhone { get; set; } = string.Empty;
    public string CustomerFirstName { get; set; } = string.Empty;
    public string CustomerLastName { get; set; } = string.Empty;
    public Guid? CustomerId { get; set; }

    // Addresses
    public string ShippingAddress { get; set; } = string.Empty; // JSON
    public string BillingAddress { get; set; } = string.Empty; // JSON

    // Totals
    public decimal SubTotal { get; set; }
    public decimal TaxAmount { get; set; }
    public decimal ShippingAmount { get; set; }
    public decimal DiscountAmount { get; set; }
    public decimal Total { get; set; }

    // Payment & Status
    public string PaymentMethod { get; set; } = "Cash"; // Cash on Delivery
    public string Status { get; set; } = "pending"; // pending, confirmed, processing, shipped, delivered, cancelled, refunded
    public string Notes { get; set; } = string.Empty;

    // Timestamps
    public DateTime CreatedAt { get; set; } = DateTime.UtcNow;
    public DateTime UpdatedAt { get; set; } = DateTime.UtcNow;

    // Navigation
    public ICollection<OrderItem> Items { get; set; } = new List<OrderItem>();
}
